### DISCUSSION TITLE
Typed action with typescript

### DISCUSSION THREAD

**Discussion by bennyKY** [statelyai/xstate#1959] [Q&A] [answered]
Hi, assuming i have this actions:
```javascript
active: {
      on: {
        ONBOARD: {
          actions: [
            { type: 'sendSms', kind: 'welcome' }
      ]
```

and in the `actions` sections:
```javascript
actions: {
  sendSms: (context, event, { action }) => actions.sendSms(context.user, action['kind'], event),
}
```

The `action` variable in `sendSms` action is not familiar with `kind` entry. 
So typescript wont have error if we'll do `actions['foo']`...
Is there a way to have it typed?

**Comment by Andarist** [comment, marked as answer]
Not yet - but we are thinking about it. We had already 2 PRs for that but we have dropped them for now - gonna revisit the topic in the future.

**Reply by bennyKY** [reply]
Thanks!
